lib/utils/Thing.js

```diff
@@ -218,8 +245,28 @@ export default class Thing {
 		return 'reddit.com';
 	}
 
+	getPostDomainUrl() {
+		const link = this.getPostDomainLink();
+		if (link) {
+			return link.href;
+		}
+		return `/domain/${this.getPostDomain()}/`;
+	}
+
 	getPostDomainLink() {
-		return this.thing.querySelector('.domain > a');
+		return this.thing.querySelector('.domain a');
+	}
+
+	getPostDomainText() {
```

Comment:
Yeah, I think I'll just stick with some minor duplication. Even if that logic changes a little, it's all tidily grouped together.